components/WorkoutsList/index.js

```diff
@@ -2,15 +2,61 @@ import Link from "next/link";
 import WorkoutForm from "../WorkoutForm";
 import WorkoutPreview from "../WorkoutPreview";
 import styled from "styled-components";
+import ModalDelete from "../ModalDelete";
+import { useState } from "react";
+
+export default function WorkoutsList({
+  onAddWorkout,
+  onDeleteWorkout,
+  workouts,
+  exercises,
+}) {
+  const [modalDelete, setModalDelete] = useState(false);
+  const [workoutIdToDelete, setWorkoutIdToDelete] = useState(0);
+  const [isWorkoutDeleted, setIsWorkoutDeleted] = useState(false);
+
+  const handleDelete = (event) => {
+    const id = event.target.dataset.workoutid; // data-workoutid auslesen
+    setWorkoutIdToDelete(id); // id speichern
+    setModalDelete(true);
+  };
+
+  function handleDeleteCancel() {
+    setModalDelete(false);
+  }
+
+  function handleDeleteConfirm() {
+    onDeleteWorkout(workoutIdToDelete);
+    setModalDelete(false);
+    setIsWorkoutDeleted(true);
+  }
 
-export default function WorkoutsList({ onAddWorkout, workouts, exercises }) {
   return (
     <>
+      {workouts.length != 0 && isWorkoutDeleted && (
+        <SuccessMessage>Workout deleted successfully!</SuccessMessage>
+      )}
```

Comment:
To handle the success message you can think about a toast message. This solution isn't really nice because this message still appears after a user first deletes a workout and then creates a new one.